,github/workflows/cicd.yaml

```
@@ -0,0 +1,31 @@
+name: ci
+
+on: push
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Install dependencies
+        run: npm install
+      - name: Run test
+        run: npm test
+  cd:
+    runs-on: ubuntu-latest
+    needs: ci
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - uses: appleboy/ssh-action@master
+        with: 
+          host: ${{ secrets.SSH_HOST }}
+          username: ${{ secrets.SSH_USERNAME }}
+          password: ${{ secrets.SSH_PASSWORD }}
+          port: ${{ secrets.SSH_PORT }}
+          script: |
+            cd ~/se2/5/se2-backend
+            git reset --hard origin/main
+            git pull https://${{ secrets.CLONE_TOKEN }}@github.com/SoftwareEngineering2-Assignment/se2-backend-group5 main
+            bash -ci 'npm install'
+            bash -ci 'pm2 restart se2-backend-group5'
```

I don't think you have to spawn a new bash process for this. 
```suggestion
            npm ci
            pm2 restart se2-backend-group5
```